File: v10/ap.cpp
#include <cstdlib>
#include <iostream>
#include <stdio.h>
#include <cmath>
using namespace std;

int main() {
    int num;
    while(cin >> num) {
        int sum = 1;
        int upperBound = sqrt(num);
        cout << upperBound << endl;
        for(int lowerBound = 2; lowerBound <= upperBound; lowerBound++) {
            if (num % lowerBound == 0) {
                if(lowerBound != num/lowerBound) {
                    sum = sum + lowerBound + num/lowerBound;   
                }
                else sum = sum + lowerBound;             
            }
        }
        cout << sum << endl;
        if (sum == num) cout << num << " perfect" << endl;
        else if (num - sum <= 2 && num - sum >= -2) cout << num << " almost perfect" << endl;
        else cout << num << " not perfect" << endl;
    }
    return 0;
}

File: v10/ap.rs
use std::io::{self, BufRead};

fn main() {
    let lines: Vec<i64> = io::stdin()
        .lock()
        .lines()
        .map(|l| l.unwrap().trim().parse().unwrap())
        .collect();

    for line in lines {
        let mut sum = 1;
        let upper = (line as f64).sqrt().floor() as i64;
        for div in 2..=upper {
            if line % div == 0 {
                if div != (line / div) {
                    sum = sum + div + (line / div);
                } else {
                    sum = sum + div;
                }
            }
        }

        print!("{line} ");

        if sum == line {
            println!("perfect");
        } else if line - sum <= 2 && line - sum >= -2 {
            println!("almost perfect");
        } else {
            println!("not perfect");
        }
    }
}
